client/state/jetpack-settings/modules/reducer.js

```diff
@@ -39,6 +50,14 @@ const createRequestsReducer = ( data ) => {
 	};
 };
 
+const createModuleListRequestReducer = ( data ) => {
```

Comment:
If we're not planning to use this helper for anything else, we could inject `isFetchingModuleList` in there and accept just its boolean value as an argument:

```
const createModuleListRequestReducer = ( isFetchingModuleList ) => {
	return ( state, { siteId } ) => {
		return merge( {}, state, {
			[ siteId ]: {
				isFetchingModuleList
			}
		} );
	};
};
```

and then change the use in the  `requests` reducer to be shorter:

```
[ JETPACK_MODULE_LIST_REQUEST ]: createModuleListRequestReducer( true ),
[ JETPACK_MODULE_LIST_REQUEST_FAILURE ]: createModuleListRequestReducer( false ),
[ JETPACK_MODULE_LIST_REQUEST_SUCCESS ]: createModuleListRequestReducer( false )
```